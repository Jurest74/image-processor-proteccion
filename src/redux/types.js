export const types = {
    imageUpload: 'imageUpload'
};
