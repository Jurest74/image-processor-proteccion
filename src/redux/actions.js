import {types} from './types';

export const newImageUploaded = (src, status, originalHeight, originalWidth, orientation) => ({
    type: types.imageUpload,
    state: {
        image: src,
        status: status,
        originalHeight: originalHeight,
        originalWidth: originalWidth,
        orientation: orientation
    }
});