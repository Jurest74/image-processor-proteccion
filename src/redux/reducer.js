import {types} from './types'

const initialState = {
    state: {
        image: null,
        status: null,
        originalHeight: null,
        originalWidth: null,
        orientation: 'vertical',
        ratio: 0
    }
};

export const imageProcessorReducer = (state = initialState, action = {}) => {
    if(action.type === types.imageUpload) {
            return {
                state: {
                    image: action.state.image,
                    status: action.state.status,
                    originalHeight: action.state.originalHeight,
                    originalWidth: action.state.originalWidth,
                    orientation: action.state.orientation,
                    ratio: action.state.ratio
                },
            };
    }else{
        return state;
    }
};