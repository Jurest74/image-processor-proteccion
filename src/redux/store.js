import {createStore} from "redux";
import {imageProcessorReducer} from "./reducer";


export const store = createStore(imageProcessorReducer);

