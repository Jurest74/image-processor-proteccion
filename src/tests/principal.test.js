import renderer from 'react-test-renderer';
import A4Sheet from '../component/A4-sheet.component.jsx';
import UploadImage from '../component/upload-image.jsx';
import {store} from '../redux/store';
import { Provider } from 'react-redux';

describe('pruebas en A4Sheet component', () =>{
    test('render A4Sheet component', () => {
        const component = renderer.create(
            <Provider store={store}>
                <A4Sheet
                    AdjustImage={false}
                />
            </Provider>,
        );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    })
});

describe('pruebas en UploadImage component', () =>{
    test('render UploadImage component', () => {
        const component = renderer.create(
            <Provider store={store}>
                <UploadImage />
            </Provider>,
        );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    })
})