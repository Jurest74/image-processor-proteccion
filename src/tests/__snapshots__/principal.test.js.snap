// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en A4Sheet component render A4Sheet component 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "display": "inline-block",
      "height": "1123px",
      "margin": "3%",
      "width": "796px",
    }
  }
>
  <img
    alt=""
    src={null}
    style={
      Object {
        "left": "5%",
        "position": "absolute",
        "top": "30%",
      }
    }
  />
</div>
`;

exports[`pruebas en A4Sheet page render A4Sheet Page 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "display": "inline-block",
      "height": "1123px",
      "margin": "3%",
      "width": "796px",
    }
  }
>
  <img
    alt=""
    src={null}
    style={
      Object {
        "left": "5%",
        "position": "absolute",
        "top": "30%",
      }
    }
  />
</div>
`;

exports[`pruebas en UploadImage component render UploadImage component 1`] = `
<div
  className="custom-file"
  style={
    Object {
      "display": "inline-block",
      "width": "796px",
    }
  }
>
  <input
    className="custom-file-input"
    id="customFile"
    onChange={[Function]}
    type="file"
  />
  <label
    className="custom-file-label"
    for="customFile"
  >
    Carga una imagen
  </label>
</div>
`;

exports[`pruebas en UploadImage page render A4Sheet Page 1`] = `
<div
  className="custom-file"
  style={
    Object {
      "display": "inline-block",
      "width": "796px",
    }
  }
>
  <input
    className="custom-file-input"
    id="customFile"
    onChange={[Function]}
    type="file"
  />
  <label
    className="custom-file-label"
    for="customFile"
  >
    Carga una imagen
  </label>
</div>
`;

exports[`pruebas en principal page render principal Page 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "display": "inline-block",
      "height": "1123px",
      "margin": "3%",
      "width": "796px",
    }
  }
>
  <img
    alt=""
    src={null}
    style={
      Object {
        "left": "5%",
        "position": "absolute",
        "top": "30%",
      }
    }
  />
</div>
`;
